src/MongoDB.Bson/IO/JsonReader.cs

```diff
@@ -2077,6 +2078,23 @@ private BsonValue ParseUndefinedExtendedJson()
             return BsonMaxKey.Value;
         }
 
+        private BsonValue ParseUuidExtendedJson()
+        {
+            VerifyToken(":");
+            var uuidToken = PopToken();
+            if (uuidToken.Type != JsonTokenType.String)
+            {
+                var message = string.Format("JSON reader expected a string but found '{0}'.", uuidToken.Lexeme);
+                throw new FormatException(message);
+            }
+            VerifyToken("}");
+
+            var hexString = uuidToken.StringValue.Replace("-", "");
+            var bytes = BsonUtils.ParseHexString(hexString);
```

Comment:
Should we check the length of the bytes array here and throw a FormatException if the length is not 16?

I realize that the BsonBinaryData constructor also performs this check, but it throws an ArgumentException instead of a FormatException.

Reply:
I didn't want to duplicate logic which `BsonBinaryData` already has but after your comment I checked where we actually throw `ArgumentException`. And there's none for parsing extended json (tests have a check for `ArgumentException`, but it's for bson cases), so I've updated the code for consistency.
Done.